### ISSUE TITLE
`reportUnusedExpression` false negative on method access

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#517]
```py
foo = ""
foo # error
foo.capitalize # no error
```

maybe its accounting for descriptors or something? need to investigate
